tensor2tensor/models/transformer.py

```diff
@@ -510,7 +518,8 @@ def inner_loop(i, finished, next_id, decoded_ids, cache, log_prob):
       return i + 1, finished, next_id, decoded_ids, cache, log_prob
 
     def is_not_finished(i, finished, *_):
-      return (i < decode_length) & tf.logical_not(tf.reduce_all(finished))
+      if decode_length_decide_end==True:return i < decode_length
```

Comment:
```
if ...:
  ...
else:
  ...
```

Reply:
hi, @rsepassi  please review the recent change if it's ok to make the merge happen. Thanks a lot!